### PR TITLE
feat: v2 embed

### LINKED ISSUE #3788: 2.0 Event Types / Embed {Flow}
[View in Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=1413%3A112136)

### PR DESCRIPTION
**Pull request by agustif**
## What does this PR do?
Embed v2 Fixes #3788
[View in figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=1413%3A112136)

## Screenshots

Screenshot 2022-09-14 at 20 35 13
Screenshot 2022-09-14 at 20 35 06
Screenshot 2022-09-14 at 20 34 54
Screenshot 2022-09-14 at 20 34 37

Screenshot 2022-09-14 at 20 35 19

### CODE REVIEW

**Review by emrysal — DISMISSED** [review]
Could you also add the EmbedButton to the top nav in the event type single view? 
image

**Comment by emrysal** [comment]
Works great! found a few glitches but nothing major

Minor: Button colour is something orangy but not black as the sidebar would make you believe on initialisation.
image

Minor: Element click button is all the way to the bottom of the screen
image

**Comment by sean-brydon** [comment]
Wait navtabs exists already right? `packages/ui/v2/core/navigation/tabs/HorizontalTabs.tsx` are these any different?

File: apps/web/components/v2/ui/NavTabs.tsx
Comment (PeerRich) [inline apps/web/components/v2/ui/NavTabs.tsx:2]: why do we need lodash

Comment (emrysal) [reply apps/web/components/v2/ui/NavTabs.tsx:2]: Instead of `noop` just do `() => {}` @agustif - never ever found a good usecase for noop.

**Comment by agustif** [comment]
> Wait navtabs exists already right? `packages/ui/v2/core/navigation/tabs/HorizontalTabs.tsx` are these any different? 
> 
> 
> 
> Edit from what I can see - yes I believe they are identical 

Thanks sean haven't found them earlier, will reuse!

File: apps/web/components/v2/ui/NavTabs.tsx
Comment (agustif) [reply apps/web/components/v2/ui/NavTabs.tsx:2]: Will reuse the ones already done!

**Comment by agustif** [comment]
@emrysal 
Fixed both! Thanks for spotting!
Re-using already done horizontal-tabs so not using loadash anymore.

I think it's ready for merge, waiting for final approval.

Screenshot 2022-09-15 at 01 54 33
Screenshot 2022-09-15 at 01 54 23


How to rebuild embed: `y workspaces @calcom/embed-core run  build`

**Comment by agustif** [comment]
Added also to event-type detail page now 
image

**Comment by agustif** [comment]
@emrysal @sean-brydon it seems we loose the icon re-using v2 navtabs, should I look into adding it back? extending current component for v2? used to work on the v1 one

**Comment by hariombalhara** [comment]
There is a fix to get Embed button in place using a different approach I guess https://github.com/calcom/cal.com/pull/4426

Please review that as well @agustif. Didn't look at your approach yet.

**Comment by hariombalhara** [comment]
Screenshot 2022-09-15 at 10 18 57 AM
@Jaibles We should increase the width and height in V2 Embed as well because Preview isn't of desktop even on desktop.

File: apps/web/pages/v2/event-types/[type]/index.tsx
Comment (hariombalhara) [inline apps/web/pages/v2/event-types/[type]/index.tsx:98]: In the approach that I have pushed changes for, it is not required to pollute event-types with Embed info

File: packages/embeds/embed-core/src/embed-iframe.ts
Comment (hariombalhara) [inline packages/embeds/embed-core/src/embed-iframe.ts:255]: No restriction on this based on Plan because it is not a Team feature.

**Comment by hariombalhara** [comment]
> There is a fix to get Embed button in place using a different approach I guess https://github.com/calcom/cal.com/pull/4426

@agustif The change is now incorporated in this branch. The approach is now to not let any Embed related info outside Embed.tsx. 
So, 
- there is no need to add embed tab values in event-types tabs.
- There is no need to implement openEmbedModal, EmbedButton already does that.

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**
